docs/application-deployment.md

```diff
@@ -100,6 +100,16 @@ This will prompt you to verify the application's name before destroying it. You
 dokku --force apps:destroy APP
 ```
 
+### Renaming a deployed app
+
+You can rename a deployed app using the `apps:rename` CLI tool:
+
+```shell
+dokku apps:rename OLD_NAME NEW_NAME
+```
+
+This will copy all of your app's contents into a new app directory with the name of your choice, delete your old app, then rebuild the new version of the app and deploy it. All of your config variables, including database urls, will be preserved.
+
```

Comment:
Probably an event should be emitted for plugins to handle updating their metadata otherwise this will always be half-broken with regards to plugin config. For example dokku-postgres can tell you which database is linked with which containers and will need a way to update that.